pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -313,6 +324,30 @@ func (dsw *desiredStateOfWorld) VolumeExists(
 	return false
 }
 
+// DetermineVolumeAction determines action to be taken for pod volume
+// if the volume should be added or removed from desired state of world.
+// return values:
+//    true  -- if volume needs to be attached
+//    false -- if volume needs to be detached
+func (dsw *desiredStateOfWorld) DetermineVolumeAction(pod *v1.Pod, defaultValue bool) bool {
```

Comment:
To handle the new `keepTerminatedPodVolumes` annotation you can add a new `keepTerminatedPodVolumes` field per node to the DSW cache, populate it on node creation, and add a new `GetKeepTerminatedPodVolumesForNode(node)` method to expose it. Then wherever you call `volumehelper.IsPodTerminated(...)` first call the new method. This is a better approach because it ensures that `desired_state_of_the_world.go` remains "just a cache"--meaning it just stores state info and allows thread-safe access to it, it does not implement business logic of "what to do".